core/domain/expression_parser.py

```diff
@@ -0,0 +1,597 @@
+    that the node represents, i.e., operator, identifier, or function.
+    """
+
+    def __init__(self, children):
+        """Initializes a Node object. For ex. 'a + b' will have root node as
+        '+' and children as ['a', 'b'].
+
+        Args:
+            children: list(Node). Child nodes of the current node.
+        """
+        self.children = children
+
+
+class AdditionOperatorNode(Node):
+    """Class representing the addition operator node."""
+
+    def __init__(self, left, right):
+        """Initializes an AdditionOperatorNode object.
+
+        Args:
+            left: Node. Left child of the operator.
+            right: Node. Right child of the operator.
+        """
+        self.operator_token = '+'
+        super(AdditionOperatorNode, self).__init__([left, right])
+
+
+class SubtractionOperatorNode(Node):
+    """Class representing the subtraction operator node."""
+
+    def __init__(self, left, right):
+        """Initializes an SubtractionOperatorNode object.
+
+        Args:
+            left: Node. Left child of the operator.
+            right: Node. Right child of the operator.
+        """
+        self.operator_token = '-'
+        super(SubtractionOperatorNode, self).__init__([left, right])
+
+
+class MultiplicationOperatorNode(Node):
+    """Class representing the multiplication operator node."""
+
+    def __init__(self, left, right):
+        """Initializes an MultiplicationOperatorNode object.
+
+        Args:
+            left: Node. Left child of the operator.
+            right: Node. Right child of the operator.
+        """
+        self.operator_token = '*'
+        super(MultiplicationOperatorNode, self).__init__([left, right])
+
+
+class DivisionOperatorNode(Node):
+    """Class representing the division operator node."""
+
+    def __init__(self, left, right):
+        """Initializes an DivisionOperatorNode object.
+
+        Args:
+            left: Node. Left child of the operator.
+            right: Node. Right child of the operator.
+        """
+        self.operator_token = '/'
+        super(DivisionOperatorNode, self).__init__([left, right])
+
+
+class PowerOperatorNode(Node):
+    """Class representing the power operator node."""
+
+    def __init__(self, left, right):
+        """Initializes an PowerOperatorNode object.
+
+        Args:
+            left: Node. Left child of the operator.
+            right: Node. Right child of the operator.
+        """
+        self.operator_token = '^'
+        super(PowerOperatorNode, self).__init__([left, right])
+
+
+class IdentifierNode(Node):
+    """Class representing the identifier node. An identifier could be a single
+    latin letter (uppercase/lowercase) or a greek letter represented by the
+    symbol name.
+    """
+
+    def __init__(self, token):
+        """Initializes an IdentifierNode object.
+
+        Args:
+            token: Token. The token representing the identifier.
+        """
+        self.token = token
```

Comment:
Similarly I can't figure out where this property gets called/used. I'm not saying necessarily that you need to remove it but I'm trying to understand if it's even needed (outside tests).

If this is just for tests, then that's probably fine, though it seems like quite a lot of set up needed just for tests.